tools/meteor-services/deploy.js

```diff
@@ -321,6 +334,122 @@ function canonicalizeSite(site) {
   return parsed.hostname;
 };
 
+// Executes the poll to check for deployment success and outputs proper messages
+// to user about the status of their app during the polling process
+function executePollForDeploymentSuccess(versionId, deployPollTimeout, result) {
+  // Create a default polling configuration for polling for deploy / build
+  // In the future, we may change this to be user-configurable or smart
+  // The user can only currently configure the polling timeout via a flag
+  let pollingState = new PollingState(deployPollTimeout);
+  
+  var deploymentPollResult = buildmessage.enterJob({
+    title: "Waiting for deploy to succeed..."},
+    function() {
+      return pollForDeploy(pollingState,
```

Comment:
You could just put the initial sleep here once, and then you don't need to have the special first run logic.

Reply:
Decent point. The reason it was in there before was because there was other stuff going on (backoff, backoff restarts, etc.) and I thought it'd be useful to have it all encapsulated in one function. Not necessary anymore though :)